Source/playerinfo/PlayerInfo.cpp

```diff
@@ -0,0 +1,416 @@
+
+                if (interface != nullptr) {
+                    result = new PlayerInfo(name, interface);
+                    std::list<PlayerInfo*>::emplace_back(result);
+                    interface->Release();
+                }
+            }
+            _adminLock.Unlock();
+
+            return (result);
+        }
+
+        static Core::NodeId Connector()
+        {
+            const TCHAR* comPath = ::getenv(_T("COMMUNICATOR_PATH"));
+
+            if (comPath == nullptr) {
+#ifdef __WINDOWS__
+                comPath = _T("127.0.0.1:62000");
+#else
+                comPath = _T("/tmp/communicator");
+#endif
+            }
+
+            return Core::NodeId(comPath);
+        }
+
+        uint32_t Delete(const PlayerInfo* playerInfo, int& refCount)
+        {
+            uint32_t result(Core::ERROR_NONE);
+
+            _adminLock.Lock();
+
+            if (Core::InterlockedDecrement(refCount) == 0) {
+                std::list<PlayerInfo*>::iterator index(
+                    std::find(std::list<PlayerInfo*>::begin(), std::list<PlayerInfo*>::end(), playerInfo));
+
+                ASSERT(index != std::list<PlayerInfo*>::end());
+
+                if (index != std::list<PlayerInfo*>::end()) {
+                    std::list<PlayerInfo*>::erase(index);
+                }
+                delete const_cast<PlayerInfo*>(playerInfo);
+                result = Core::ERROR_DESTRUCTION_SUCCEEDED;
+            }
+
+            _adminLock.Unlock();
+
+            return result;
+        }
+
+    private:
+        PlayerInfo* Find(const string& name)
+        {
+            PlayerInfo* result(nullptr);
+
+            std::list<PlayerInfo*>::iterator index(std::list<PlayerInfo*>::begin());
+
+            while ((index != std::list<PlayerInfo*>::end()) && ((*index)->Name() != name)) {
+                index++;
+            }
+
+            if (index != std::list<PlayerInfo*>::end()) {
+                result = *index;
+                result->AddRef();
+            }
+
+            return result;
+        }
+
+        Core::CriticalSection _adminLock;
+        Core::ProxyType<RPC::InvokeServerType<1, 0, 8>> _engine;
+        Core::ProxyType<RPC::CommunicatorClient> _comChannel;
+    };
+    mutable int _refCount;
+    const string _name;
+    Exchange::IPlayerProperties* _playerConnection;
+    static PlayerInfo::PlayerInfoAdministration _administration;
+
+public:
+    PlayerInfo() = delete;
+    PlayerInfo(const PlayerInfo&) = delete;
+    PlayerInfo& operator=(const PlayerInfo&) = delete;
+
+    static PlayerInfo* Instance(const string& name)
+    {
+        return _administration.Instance(name);
+    }
+    void AddRef() const
+    {
+        Core::InterlockedIncrement(_refCount);
+    }
+    uint32_t Release() const
+    {
+        return _administration.Delete(this, _refCount);
+    }
+
+    const string& Name() const { return _name; }
+
+    int8_t IsAudioEquivalenceEnabled(bool* isEnabled) const
```

Comment:
here we're back in the C++ realm, so this can be a reference, so you don't have to worry for a nullptr